aerogearcatalog/keycloak-apb
# [AEROGEAR-1995] Update bind to configure when shared

## Description
Update the bind playbook to connect to the remote keycloak and perform the configuration. This should write the config to the local namespace. It should also account for custom realms.

## Related Notes
Related JIRA Ticket - https://issues.jboss.org/browse/AEROGEAR-1995

## Test Cases
- Binding with a custom realm
  - Enter in a value for the `Name of Keycloak realm` parameter in the bind dialog creation.
  - Ensure that a realm of the specified name has been created in the Keycloak web console.
  - Ensure that a client bearer of the specified service is created within that realm.
  - Ensure that the right configuration is saved in the local namespace.

- Binding with the default namespace realm
  - Complete binding process without entering any value for the `Name of Keycloak realm` parameter.
  - Ensure that a realm is created with the current namespace name.
  - Ensure that a client bearer of the specified service is created within that realm
  - Ensure that the right configuration is saved in the local namespace.

- Error Message should show in the console on client creation conflict
  - Create a client with the same service name of an existing client within the same realm.
  - Ensure that the bind fails and that proper error message is shown in the logs.

## Screenshots
![image](https://user-images.githubusercontent.com/9078522/36539157-bc847494-17cd-11e8-8c79-a02f2d9c142a.png)
